lib/MultiCompiler.js

```diff
@@ -2,175 +2,173 @@
+const MultiStats = require("./MultiStats");
 
-	function delegateProperty(name) {
-		Object.defineProperty(this, name, {
-			configurable: false,
-			get: function() {
-				throw new Error("Cannot read " + name + " of a MultiCompiler");
-			},
-			set: function(value) {
-				this.compilers.forEach(function(compiler) {
-					compiler[name] = value;
-				});
-			}.bind(this)
-		});
+module.exports = class MultiCompiler extends Tapable {
+	constructor(compilers) {
+		super();
+		if(!Array.isArray(compilers)) {
+			compilers = Object.keys(compilers).map((name) => {
+				compilers[name].name = name;
+				return compilers[name];
+			});
+		}
+		this.compilers = compilers;
+		let doneCompilers = 0;
+		let compilerStats = [];
+		this.compilers.forEach((compiler, idx) => {
+			let compilerDone = false;
+			compiler.plugin("done", stats => {
+				if(!compilerDone) {
+					compilerDone = true;
+					doneCompilers++;
+				}
+				compilerStats[idx] = stats;
+				if(doneCompilers === this.compilers.length) {
+					this.applyPlugins("done", new MultiStats(compilerStats));
+				}
+			});
+			compiler.plugin("invalid", () => {
+				if(compilerDone) {
+					compilerDone = false;
+					doneCompilers--;
+				}
+				this.applyPlugins("invalid");
+			});
+		}, this);
 	}
-	delegateProperty.call(this, "outputFileSystem");
-	delegateProperty.call(this, "inputFileSystem");
 
-	Object.defineProperty(this, "outputPath", {
-		configurable: false,
-		get: function() {
-			var commonPath = compilers[0].outputPath;
-			for(var i = 1; i < compilers.length; i++) {
-				while(compilers[i].outputPath.indexOf(commonPath) !== 0 && /[\/\\]/.test(commonPath)) {
-					commonPath = commonPath.replace(/[\/\\][^\/\\]*$/, "");
-				}
+	get outputPath() {
+		let commonPath = this.compilers[0].outputPath;
+		for(const compiler of this.compilers) {
+			while(compiler.outputPath.indexOf(commonPath) !== 0 && /[\/\\]/.test(commonPath)) {
+				commonPath = commonPath.replace(/[\/\\][^\/\\]*$/, "");
 			}
-			if(!commonPath && compilers[0].outputPath[0] === "/") return "/";
-			return commonPath;
 		}
-	});
 
-	var doneCompilers = 0;
-	var compilerStats = [];
-	this.compilers.forEach(function(compiler, idx) {
-		var compilerDone = false;
-		compiler.plugin("done", function(stats) {
-			if(!compilerDone) {
-				compilerDone = true;
-				doneCompilers++;
-			}
-			compilerStats[idx] = stats;
-			if(doneCompilers === this.compilers.length) {
-				this.applyPlugins("done", new MultiStats(compilerStats));
-			}
-		}.bind(this));
-		compiler.plugin("invalid", function() {
-			if(compilerDone) {
-				compilerDone = false;
-				doneCompilers--;
-			}
-			this.applyPlugins("invalid");
-		}.bind(this));
-	}, this);
-}
-module.exports = MultiCompiler;
+		if(!commonPath && this.compilers[0].outputPath[0] === "/") return "/";
+		return commonPath;
+	}
 
-MultiCompiler.prototype = Object.create(Tapable.prototype);
-MultiCompiler.prototype.constructor = MultiCompiler;
+	get inputFileSystem() {
+		return {
```

Comment:
This should be:

``` js
get inputFileSystem() {
  throw new Error("Cannot read inputFileSystem of a MultiCompiler");
}
```

The object with `configurable: false, get: () => ...` is the argument object of Object.defineProperty.